File: dataround-link-web/src/api/virtualtable.ts
/*
 * Copyright (C) 2025 yuehan124@gmail.com
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

/**
 * @description: api
 * @author: yuehan124@gmail.com
 * @date: 2026-06-05
 */
import { http, baseAPI } from "../utils";

export const getVirtualTableById = (id: string) => {
  return http.get(baseAPI + "/vtable/" + id);
};

export const getVirtualTableList = () => {
  return http.get(baseAPI + "/vtable/list");
};

export const saveOrUpdateVirtualTable = (vtable: any) => {
  return http.post(baseAPI + "/vtable/saveOrUpdate", vtable);
}

export const deleteVirtualTable = (id: string) => {
  return http.delete(baseAPI + "/vtable/" + id);
}

File: dataround-link-svc/src/main/java/io/dataround/link/entity/VirtualTable.java
/*
 * Copyright (C) 2025 yuehan124@gmail.com
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package io.dataround.link.entity;

import java.util.Date;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;

/**
 * Entity class representing a virtual table in the system.
 * Contains metadata about a table including its database, name, configuration and associated fields.
 *
 * @author yuehan124@gmail.com
 * @date 2025-05-06
 */
@Data
@TableName("virtual_table")
public class VirtualTable {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;
    private Long connectionId;
    private Long projectId;
    @TableField("\"database\"")
    private String database;
    private String tableName;
    private String tableConfig;
    private String description;
    private Boolean deleted;
    private Long createBy;
    private Long updateBy;
    private Date createTime;
    private Date updateTime;
}
